src/components/substance/SubstanceSidebar.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <div>
+    <b-button v-b-toggle.substance-sidebar class="btn btn-primary btn-sm">
+      Toggle Sidebar
+    </b-button>
+    <b-sidebar id="substance-sidebar" shadow>
+      <SubstanceTree :items="treeData" />
+    </b-sidebar>
+  </div>
+</template>
+
+<script>
+import SubstanceTree from "@/components/substance/SubstanceTree";
+
+export default {
+  name: "SubstanceSidebar",
+  components: {
+    SubstanceTree
+  },
+  computed: {
+    treeData: function() {
+      let data = this.$store.state.substance.list;
+      let included = this.$store.state.substance.included;
+      // create an array of date objects with nested children
+      let dates = [];
+      data.forEach(substance => {
+        let dt = substance.attributes.updatedAt;
+        let dateString;
+        let user;
+        if (!dt) {
+          dateString = "Not Tracked";
+        } else {
+          dateString = new Date(dt).toISOString().split("T")[0];
+        }
+        let userData = substance.relationships.updatedBy.data;
+        console.log(substance.relationships.updatedBy);
+        if (!userData) {
+          user = "Unknown User";
+        } else {
+          user = included.user[userData.id].attributes.username;
+        }
+        let dateObj = dates.filter(o => Object.keys(o).includes(dateString));
+        if (dateObj.length < 1) {
+          dateObj = { name: dateString, icon: "calendar", children: [] };
+          dates.push(dateObj);
+        }
+        let userObj = dateObj.children.filter(u =>
+          Object.keys(u).includes(user)
+        );
+        if (userObj.length < 1) {
+          userObj = { name: user, icon: "person-fill", children: [] };
+        }
+        userObj.children.push({
+          id: substance.id,
+          name: substance.attributes.sid,
+          icon: "pencil-square"
+        });
+        dateObj.children.push(userObj);
+      });
```

Comment:
I've made a few minor changes to how you're generating this tree.  I like this function a lot better.  I went ahead and removed the console log and made some other minor changes.

- Filtering an array returns and array.  You were referring to these as objects later on.  Shifting to return the first result if one is found should solve this.    
- You were filtering keys and not values.
- You were pushing **user** objects regardless of whether they were new or old.
- I also added some comments to break the sections apart visually.  

See how you like the look of this:
```suggestion
      data.forEach(substance => {
        let dt = substance.attributes.updatedAt;
        let dateString;
        let user;

        // Get Date String
        if (!dt) {
          dateString = "Not Tracked";
        } else {
          dateString = new Date(dt).toISOString().split("T")[0];
        }

        // Get User String
        let userData = substance.relationships.updatedBy.data;
        if (!userData) {
          user = "Unknown User";
        } else {
          user = included.user[userData.id].attributes.username;
        }

        // Add/Find Date
        let dateObj = dates.filter(o => Object.values(o).includes(dateString));
        if (dateObj.length < 1) {
          dateObj = { name: dateString, icon: "calendar", children: [] };
          dates.push(dateObj);
        } else {
          dateObj = dateObj.shift();
        }

        // Add/Find User
        let userObj = dateObj.children.filter(u =>
          Object.values(u).includes(user)
        );
        if (userObj.length < 1) {
          userObj = { name: user, icon: "person-fill", children: [] };
          dateObj.children.push(userObj);
        } else {
          userObj = userObj.shift();
        }

        // Add Substance
        userObj.children.push({
          id: substance.id,
          name: substance.attributes.sid,
          icon: "pencil-square"
        });
      });
```

Reply:
Github is doing a really bad diff on this suggestion but hopefully you can see the changes I've made, and you think they make sense.

Reply:
cool, I had to commit these just to see what you had done, but I like it